src/libopensc/card-iasecc.c

```diff
@@ -594,6 +609,36 @@ iasecc_init_amos_or_sagem(struct sc_card *card)
 }
 
 
+static int
+iasecc_init_cpx(struct sc_card *card)
+{
+	struct sc_context *ctx = card->ctx;
+	unsigned int flags; /* TBC it is not IASECC_CARD_DEFAULT_FLAGS */
+
+	LOG_FUNC_CALLED(ctx);
+
+	card->caps = SC_CARD_CAP_RNG; /* TBC it is not IASECC_CARD_DEFAULT_CAPS */
+
+	flags = SC_ALGORITHM_RSA_PAD_PKCS1;
+	flags |= SC_ALGORITHM_RSA_RAW;
+
+	/* No signature with contactless mode */
+	if (card->type == SC_CARD_TYPE_IASECC_CPXCL)
+		flags |= SC_ALGORITHM_RSA_HASH_NONE;
+	else
+		flags |= SC_ALGORITHM_RSA_HASH_SHA1    |
+		         SC_ALGORITHM_RSA_HASH_SHA256;
+
+	_sc_card_add_rsa_alg(card, 1024, flags, 0);
+	_sc_card_add_rsa_alg(card, 2048, flags, 0);
+	_sc_card_add_rsa_alg(card, 512, flags, 0);
```

Comment:
nit: would be more readable it the values 1024, 2048 and 512 would be ordered, either ascending or descending.

Reply:
done: https://github.com/OpenSC/OpenSC/pull/2217/commits/3867bd357a9122f78aa92bfa4a49c584e4e47b7c